expression/aggregation/window_func.go

```diff
@@ -27,30 +27,34 @@ type WindowFuncDesc struct {
 }
 
 // NewWindowFuncDesc creates a window function signature descriptor.
-func NewWindowFuncDesc(ctx sessionctx.Context, name string, args []expression.Expression) *WindowFuncDesc {
+func NewWindowFuncDesc(ctx sessionctx.Context, name string, args []expression.Expression) (*WindowFuncDesc, error) {
 	switch strings.ToLower(name) {
 	case ast.WindowFuncNthValue:
 		val, isNull, ok := expression.GetUint64FromConstant(args[1])
 		// nth_value does not allow `0`, but allows `null`.
 		if !ok || (val == 0 && !isNull) {
-			return nil
+			return nil, nil
```

Comment:
I do not like a nil object with nil error, Could it be something like `return nil, someErr`?

Reply:
The outer caller will check the nil object. It's somewhat similar to a flag.
But I agree with you that return `someErr` is more acceptable.

/cc @lamxTyler

Reply:
@SunRunAway This is a cherry-pick commit. If we'll do the modification, it will be accomplished in another PR.

Reply:
Yes, we can change it in another pr.